jenkins/scripts/VersionSelectorScript.groovy

```diff
@@ -75,8 +75,6 @@ def buildExclusions = [
   // OSX ---------------------------------------------------
   [ /osx1015/,                        anyType,     gte(21) ],
   [ /osx11-x64-release-tar/,          releaseType, lt(20)  ],
-  [ /osx1015-release-pkg/,            releaseType, gte(16) ],
-  [ /osx1015-release-tar/,            releaseType, gte(20) ],
```

Comment:
We'll need to adjust `osx11-x64-release-tar` so that Node.js 18 runs on those instead of `osx1015-release-tar` otherwise we'll have no Node.js 18 macOS x64 tarballs.

Reply:
Not sure how to do that, can you guide me or provide an example? I was using an old PR as a reference for this change 🤔

Reply:
If only one `release-tar` remains, you can remove the previous line, which currently excludes Node.js 18 from `osx11-x64-release-tar`.
Also, if there are no longer any `osx1015` configs, you can delete the first line.
In other words, if after this change all Node.js versions are supposed to be built on the same machine label, you can remove all osx-related lines from this file.

Reply:
@UlisesGascon Quick overview of how this works:
- Job has all possible things it might run on added as configuration matrix selections.
- Job also has reference to this groovy script which places addition restrictions. The table at the top of this file lists where to exclude (i.e. do not run) configurations matching the regular expressions. If nothing excludes the configuration being matched against (i.e. no entry in this table) it is allowed.

As an aside, please don't merge/make any change to the set up until after tomorrow's Node.js 20 LTS transition release is done.

Reply:
Thanks for the explanations and the feedback @targos @richardlau 👍 

I made some changes in 146d9cf:
- I removed the references to `osx1015`
- I added support for Node18 in `osx11-x64-release-tar`
- I added `osx11-x64-release-pkg` so the pkg notarization can be done also (not sure if this is correct). As far I know we want to target pkg rather than tarball currently.